### PR TITLE
Fix inconsistent indentation for ternary in call expressions

### LINKED ISSUE #10265: Inconsistent indentation for ternary in function call
From: https://github.com/prettier/prettier/pull/10187#issuecomment-774759066

**Prettier pr-10187**
[Playground link](https://deploy-preview-10187--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAjAlgMwgJygYR3SgHM4cBZCAZwAs4ATABXIAcAbOADwGUBDAO4NyAAgC8IzFAAUAHSgiR0ku3QBbanADi7AJ6ta1ABJ8AXnxwMAchAitqAFXRwnccWIkAGeYsUB+ET4AN3IcRgAhPhh6HGpwiC4cWzVwgFcSEl0rVJx2H18kQKgIBl0oOHxoBhoKPnYouAB5EJwGgCU4GGIIPIURAEoAOlYk1nl+gG55eSk5PuVVDW09A2MzC2tbe1dXdy98-0CWsIZI6PI4hKSIFPTM7NyDkUK+YtLyyqhq6lr6+GbyO1Ot1eoohiM7OMplAQAAaEB2LrQajIUAWJICJgWBAolB1AR8XQo+GoVpgADWnR4rD4YGIJGQMBwqTg8LgalQjGEDAAMq8SKk+GQAGK4NRRLqkZAgPipGAQOEgWgwNTsADqtHQ8GoNLAcB4OK16CCWt00rA1GJIGImhwMCYrRI4uQmDqmnhACtqFxwmTKTB+Go4DziHAXW7WSAvbx6ZwAIqpCDwcPsd0gGmxcjS1B8TnsRUjYgwNXoBjRZAADk88IhmjVrVY0pGcFtIUVAEdE-AHXZcTLqABacqMRiKsKd9BhB1C51IV2pyOaNToRnMxexuAJpNhucR+EwXMlsu0ZAAJn3rXQqlIlTUs5ALYArIrUpoHLncfO00EWQBJL4IAGYBEKwMAAIJfDwMC6JwKaaAAvvBQA)
```sh
--parser babel
```

**Input:**
```jsx
bifornCringerMoshedPerplexSawder = fn(
  (glimseGlyphsHazardNoopsTieTie === 0
    ? averredBathersBoxroomBuggyNurl
    : anodyneCondosMalateOverateRetinol
  ).prop
);

fn(
  (glimseGlyphsHazardNoopsTieTie === 0
    ? averredBathersBoxroomBuggyNurl
    : anodyneCondosMalateOverateRetinol
  ).prop
);

```

**Output:**
```jsx
bifornCringerMoshedPerplexSawder = fn(
  (
    glimseGlyphsHazardNoopsTieTie === 0
      ? averredBathersBoxroomBuggyNurl
      : anodyneCondosMalateOverateRetinol
  ).prop
);

fn(
  (glimseGlyphsHazardNoopsTieTie === 0
    ? averredBathersBoxroomBuggyNurl
    : anodyneCondosMalateOverateRetinol
  ).prop
);

```

**Expected behavior:**

```jsx
bifornCringerMoshedPerplexSawder = fn(
  (glimseGlyphsHazardNoopsTieTie === 0
    ? averredBathersBoxroomBuggyNurl
    : anodyneCondosMalateOverateRetinol
  ).prop
);

fn(
  (glimseGlyphsHazardNoopsTieTie === 0
    ? averredBathersBoxroomBuggyNurl
    : anodyneCondosMalateOverateRetinol
  ).prop
);
```

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Fixes #10265

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
Same thing in MemberExpression, 

**Prettier pr-10266**
[Playground link](https://deploy-preview-10266--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAjAlgMwgJygYR3SgHM4cBZCAZwAs4ATABXIAcAbOADwGUBDAO4NyAAgC8IzFADaAHSgiRAChLt0AW2pwA4uwCerWtQASfAF58cDAHIQIragBV0cZ3HFiJABnmLFAfhE+ADdyHEYAIT4YehxqCIguHDt1CIBXEhI9azScdl8-JCCoCAY9KDh8aAYaCj52aLgAeVCcRoAlOBhiCHyFEQBKADpWZNZ5AF15aagpOX6VNU0dfUMTc0sbOwc3Nw9vAoCg1vCGKJjyeMTkiFSMrJy8w5EivhKyiqqoGuo6hvgWuQOl0en1FMNRvZJgBuGYgAA0IHs3Wg1GQoEsyQETEsCDRKHqAj4ejRiNQbTAAGsujxWHwwMQSMgYDg0nBEXB1KhGMIGAAZN4kNJ8MgAMVw6mi3VIyBAfDSMAgCJAtBg6nYAHVaOh4NQ6WA4Dw8Tr0MEdXpZWBqKSQMQtDgYEw2iRJchMPUtIiAFbULgRCnUmD8dRwPnEOBuj3skA+3iMzgARTSEHgkfYnpAdLi5FlqD43PYytGxBgGvQDBiyAAHF5EZCtBq2qxZaM4PbQsqAI7J+BO+z4uXUAC0FUYjGV4W76HCTpFrqQ7vT0a06nQzNZy-jcCTKYjC6jiJg+bLFdoyAATIe2ug1KQqup5yA2wBWZVpLSOfP4xcZ4JsgCS3wIEGYBEKwMAAILfDwMB6JwaZaAAvohQA)
```sh
--parser babel
```

**Input:**
```jsx
bifornCringerMoshedPerplexSawder = fn[
  (glimseGlyphsHazardNoopsTieTie === 0
    ? averredBathersBoxroomBuggyNurl
    : anodyneCondosMalateOverateRetinol
  ).prop
]


fn[
  (glimseGlyphsHazardNoopsTieTie === 0
    ? averredBathersBoxroomBuggyNurl
    : anodyneCondosMalateOverateRetinol
  ).prop
];


```

**Output:**
```jsx
bifornCringerMoshedPerplexSawder =
  fn[
    (
      glimseGlyphsHazardNoopsTieTie === 0
        ? averredBathersBoxroomBuggyNurl
        : anodyneCondosMalateOverateRetinol
    ).prop
  ];

fn[
  (glimseGlyphsHazardNoopsTieTie === 0
    ? averredBathersBoxroomBuggyNurl
    : anodyneCondosMalateOverateRetinol
  ).prop
];

```

File: src/language-js/print/ternary.js [hunk 1]
```diff
@@ -174,14 +174,23 @@ function shouldExtraIndentForConditionalExpression(path) {
...
     ancestor = path.getParentNode(++ancestorCount);
-  }
+    const getAncestorName = () =>
+      path.callParent((path) => path.getName(), ancestorCount - 1);
+    if (
+      ancestor == null ||
+      (isCallExpression(ancestor) && getAncestorName() !== "callee") ||
+      (isMemberExpression(ancestor) &&
+        ancestor.computed &&
+        getAncestorName() !== "object")
```
Comment (fisker) [inline src/language-js/print/ternary.js:188]: I think we can use the same condition on line 211

**Review by thorn0 — APPROVED** [review]
Looks good.

File: src/language-js/print/ternary.js (hunk 1 above)
Comment (fisker) [reply src/language-js/print/ternary.js:188]: I'll try

Comment (fisker) [reply src/language-js/print/ternary.js]: Check https://github.com/prettier/prettier/pull/10266/commits/a32e498ecb4c0bdca381caa81296959c8a4a0da0, I think it's more clear

**Comment by fisker** [comment]
I don't use TS, should we indent this case too?

**Prettier pr-10266**
[Playground link](https://deploy-preview-10266--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuc0DOMAEMCeAHOAZTACcBLPGAQTQFEAPPEuNNM6TAXkwAoewAhgBshAOQgATOADo8A5rAD803AS6dOAHShVCAFQD6ogPIARWgb0BNAAq1C0gMIALAWSgMmLNtG2ZMipiCIuJSsvIIMOEKMH6YSEHCYpIycjEAlJgCaJh6hPZ6zDIAQu7yOJ7MrOxQ6Sr4RKQU1HSMVT5QANza2pBQGNgNxOSUNJXeNVy8-EmhqRFK9WoaWjr6RmYW1nYOLm4ebRO+UP6BwclhaZHRkXEJ53M3sJnZufn6RdKlUOXj1dDpHiqRojFp-DrpTogAA0IAglBqaGQoHkJAgAHcbAskShhOiBDgkbCAEYkARgADWcBghDkYHcAHNkDASABXOCwuAAW2JcAkUgkABkBFAGayBAy4AAxCAkLkCGAwRnIEACVkwCAwkDOGBcoQAdWcZHgaDpRAQbCVADdjTgVWBWFr3Gg4CQYDYyQz5cgAGbCF2wgBWaHoxTJlOphAEXLggvccF9-o5IGD9EIjKEcAAiqyIPBE0IAyA0i6SCrgWgmpQtUx3DB9WQJDBnMgABwABlhTAgLv1ZLwKq8patCdhAEdc-APfCcaq0ABaKBwPl8rXMCdkZgeiXepB+wvJl1csjMtmHjPZycJvdJ2EwATEhtNltIABMd7JZCEjMcEC5u5AFgAFYtVZF09AfHF9yLK12QASSgKRYGGZoqEQwhcEzAsXQAXxwoA)
```sh
--parser typescript
```

**Input:**
```tsx
const typeScriptAsExpression = ((callNode.parent?.type ===
AST_NODE_TYPES.ChainExpression
  ? callNode.parent.parent
  : callNode.parent) as TSESTree.BinaryExpression).typeScriptAsExpression;

const typeScriptAsExpression = ((callNode.parent?.type ===
AST_NODE_TYPES.ChainExpression
  ? callNode.parent.parent
  : callNode.parent) as TSESTree.BinaryExpression)(typeScriptAsExpression);
```

**Output:**
```tsx
const typeScriptAsExpression = ((callNode.parent?.type ===
AST_NODE_TYPES.ChainExpression
  ? callNode.parent.parent
  : callNode.parent) as TSESTree.BinaryExpression).typeScriptAsExpression;

const typeScriptAsExpression = ((callNode.parent?.type ===
AST_NODE_TYPES.ChainExpression
  ? callNode.parent.parent
  : callNode.parent) as TSESTree.BinaryExpression)(typeScriptAsExpression);

```

If we do, we only need keep looking for parent of `TSAsExpression`

**Comment by thorn0** [comment]
Let's merge?

File: src/language-js/print/ternary.js (hunk 1 above)
Comment (sosukesuzuki) [reply src/language-js/print/ternary.js]: Thank you!

**Pull request merged**